okex-client/src/client/mod.rs

```diff
@@ -347,21 +347,14 @@ impl OkexClient {
             .send()
             .await?;
 
-        let PositionData {
-            notional_usd, pos, ..
-        } = Self::extract_response_data::<PositionData>(response).await?;
+        let PositionData { notional_usd, .. } =
+            Self::extract_response_data::<PositionData>(response).await?;
 
-        let direction = pos.parse::<Decimal>().unwrap_or(Decimal::ZERO);
```

Comment:
This is dangerous - if we don't distinguish between exposure direction the dealer might be in a completely opposite state and the system would not know it!

Reply:
Right and if possible to happen, I thought we should validate it at the source instead of all the clutter it generates downstream.

Reply:
It is possible to happen and the business logic has to be aware of it. How else can it make the right decisions if it doesn't know what state the position is in?

Reply:
> Right and if possible to happen, I thought we should validate it at the source instead of all the clutter it generates downstream.

Let me clarify something here `validation` is something we do to input that we can reject. EG if a user provides us a value that is out of the range of the bounds of what we accept we can and should validate it and reject if it doesn't fulfil our implicit contract.
This value is not something we can validate as it is not something we can reject - it represents external state that simply _is what it is_ and we have to deal with it no matter what.